dataforge-distributed/src/jvmMain/kotlin/space/kscience/dataforge/distributed/ServiceWorkspace.kt

```diff
@@ -0,0 +1,119 @@
+package space.kscience.dataforge.distributed
+
+import io.ktor.utils.io.core.*
+import io.lambdarpc.dsl.LibService
+import io.lambdarpc.dsl.def
+import io.lambdarpc.dsl.j
+import io.lambdarpc.utils.ServiceId
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.KSerializer
+import space.kscience.dataforge.context.Context
+import space.kscience.dataforge.context.Global
+import space.kscience.dataforge.context.gather
+import space.kscience.dataforge.data.DataSet
+import space.kscience.dataforge.data.DataTree
+import space.kscience.dataforge.distributed.serialization.DataSetPrototype
+import space.kscience.dataforge.distributed.serialization.MetaCoder
+import space.kscience.dataforge.distributed.serialization.NameCoder
+import space.kscience.dataforge.distributed.serialization.TaskRegistryCoder
+import space.kscience.dataforge.meta.Meta
+import space.kscience.dataforge.names.Name
+import space.kscience.dataforge.names.asName
+import space.kscience.dataforge.workspace.SerializableResultTask
+import space.kscience.dataforge.workspace.Task
+import space.kscience.dataforge.workspace.TaskResult
+import space.kscience.dataforge.workspace.Workspace
+import space.kscience.dataforge.workspace.wrapResult
+
+/**
+ * Workspace that exposes its tasks for remote clients.
+ * @param port Port to start service on. Will be random if null.
+ */
+public class ServiceWorkspace(
+    port: Int? = null,
+    override val context: Context = Global.buildContext("workspace".asName()),
+    private val dataSerializer: KSerializer<Any>? = null,
+    data: DataSet<*> = runBlocking { DataTree<Any> {} },
```

Comment:
DataTree builder will be non-suspending in the next version, but it should be possible to create it without suspending now.